File: regress/rust/json-password.rs
use orb::ort;
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    assert_eq!(args.len(), 2);
    let db = ort::Ortdb::new_with_args(&args[1], ort::Ortargs { bcrypt_cost: 4 });
    let ctx = db.connect().unwrap();

    let id1 = ctx.db_foo_insert(&"testing".to_string(), &"foobarbaz".to_string(), ort::data::Enm::A, 0x01).unwrap();

    let obj1 = ctx.db_foo_get_id(id1).unwrap().unwrap();

    assert_eq!(obj1.export(), r#"{"a":"testing","c":"200","d":"1","id":"1"}"#);


}

File: regress/nodejs/json-password.ts

const db: ortdb = ort(dbfile);
const ctx: ortctx = db.connect();

const id1: bigint = ctx.db_foo_insert('testing', 'foobarbaz', ortns.enm.a, BigInt(0x01));
if (id1 < 0)
	return false;

const obj1: ortns.foo|null = ctx.db_foo_get_id(id1);
if (obj1 === null)
	return false;

if (JSON.stringify(obj1.export()) !== '{"a":"testing","c":"200","d":"1","id":"1"}')
	return false;

return true;
